api/bucketsd/client/client_test.go

```diff
@@ -955,6 +957,55 @@ func pushPathAccessRoles(
 	})
 }
 
+func TestClient_AcceptPathAccessRoles(t *testing.T) {
+	ctx, userctx, threadsclient, client := setupForUsers(t)
+	buck, err := client.Create(ctx)
+	require.NoError(t, err)
+
+	q, err := client.PushPaths(ctx, buck.Root.Key)
+	require.NoError(t, err)
+	err = q.AddFile("file", "testdata/file1.jpg")
+	require.NoError(t, err)
+	for q.Next() {
+		require.NoError(t, q.Err())
+	}
+	require.NoError(t, q.Close())
+
+	// Grant Token Access
+	token, tokenHash := generateShareToken(t)
+	err = client.PushPathAccessRoles(ctx, buck.Root.Key, "file", map[string]bucks.Role{
+		tokenHash: bucks.Writer,
+	})
+	require.NoError(t, err)
+
+	// New User Accepts File
+	user2, user2ctx := newUser(t, userctx, threadsclient)
+	role, err := client.AcceptPathAccessRoles(ctx, buck.Root.Key, "file", token, user2.GetPublic().String())
+	require.NoError(t, err)
+	require.Equal(t, bucks.Writer, role)
+
+	// verify new user can access file
+	check := accessCheck{
+		Key:  buck.Root.Key,
+		Path: "file",
+		Read: true,
+		Write: true,
+	}
+	checkAccess(t, user2ctx, client, check)
+}
+
+func generateShareToken(t *testing.T) (string, string) {
```

Comment:
Okay, I'll make the update.

> However, all that should be done internally (in the service) so that the caller doesn't need to know how to construct the token.

I'm not sure what you mean by the above? Do you mean the `PushPathAccessRole` function should accept the raw token and it performs the hashing internally instead of forcing the user to perform the hashing before calling the function? @sanderpick

Reply:
I think it would be nice to hide as much complexity as possible from the user. So, this could be done in the buckets service handler, but that would require a new API. We could start by just adding a new client method like:
```
func (c *Client) PushPathAccessRoleInvite(ctx context.Context, key, pth string, role buckets.Role) (string, error) {
    // create a new token from random bytes
    // hash the token
    // push the new role:
    c.PushPathAccessRoles(ctx, key, pth, map[string]buckets.Role{hash: role})
    // return the raw token so it can be given back to the caller to do whatever they want with it
    return token
}
```